app/models/v2/report.rb

```diff
@@ -44,12 +86,25 @@ def percent_compliant
     has_many :report_systems, class_name: 'V2::ReportSystem', dependent: nil # rubocop:disable Rails/InverseOf
     has_many :systems, class_name: 'V2::System', through: :report_systems
     has_many :reported_systems, class_name: 'V2::System', through: :test_results, source: :system, dependent: nil
+    # rubocop:disable Rails/InverseOf
+    has_many :reporting_and_non_reporting_systems, lambda {
+      joins(
+        arel_table.join(V2::System.arel_table, AN::InnerJoin)
+                  .on(V2::System.arel_table[:id].eq(V2::ReportSystem.arel_table[:system_id]))
+                  .join(V2::TestResult.arel_table, AN::OuterJoin)
+                  .on(V2::TestResult.arel_table[:system_id].eq(V2::System.arel_table[:id]))
+                  .join_sources
+      ).where(
+        Pundit.policy_scope(User.current, V2::System).where_clause.ast.to_sql
+      )
+    }, class_name: 'V2::ReportSystem', dependent: nil
```

Comment:
Why is this pointing to the `ReportSystem` model? What are we exactly joining here? Not the `System` and the `TestResult` model? We should pick one of those, but the semantically correct is the `System` as not all records will have a `TestResult`.